pkg/middleware/auth.go

```diff
@@ -0,0 +1,65 @@
+package middleware
+
+import (
+	"net/http"
+
+	clouddriver "github.com/billiford/go-clouddriver/pkg"
+	"github.com/billiford/go-clouddriver/pkg/fiat"
+	"github.com/gin-gonic/gin"
+)
+
+//authApplication takes a list of permissions
+//authAccount
+//go:generate counterfeiter . Auth
+type Auth interface {
+	AuthApplication() gin.HandlerFunc
+}
+
+type permissions struct {
+	list []string
+}
+
+func NewPermissions(list []string) permissions {
+	return permissions{list: list}
+}
+
+func (p *permissions) AuthApplication() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		user := c.GetHeader("user")
+		app := c.GetHeader("application")
+
+		if len(user) == 0 || len(app) == 0 {
```

Comment:
string checks in the rest of the application are done using `str == ""` instead of checking the length, so we should do the same here.